src/forms/TimeInput.jsx

```diff
@@ -78,321 +78,326 @@ export class TimeInput extends React.Component {
 	* 	time input chages. In turn calls the onChange handler prop, if there is one provided
 	* 	(eg supplied by redux-form or DateTimePicker) with the updated values
 	*/
-	onChange(partialState) {
-		if (!this.props.onChange) {
-			return;
-		}
-		const stateValues = {
-			...this.state,
-			...partialState
-		};
-		// value in state needs to be updated
-		const value = `${formatHours(stateValues.hours, stateValues.meridian)}:${formatDigits(stateValues.minutes)}`;
-		this.props.onChange(value);
-		this.props.onChangeCallback && this.props.onChangeCallback();
-	}
+  onChange(partialState) {
+    if (!this.props.onChange) {
+      return;
+    }
+    const stateValues = {
+      ...this.state,
+      ...partialState
+    };
+    // value in state needs to be updated
+    const value = `${formatHours(stateValues.hours, stateValues.meridian)}:${formatDigits(stateValues.minutes)}`;
+    this.props.onChange(value);
+    this.props.onChangeCallback && this.props.onChangeCallback();
+  }
 
 	/**
 	* @function onTimeInputChange
 	* @param e Event Object
 	* @description called when the browser-native time input changes, in turn calls the onChange
 	* 	handler prop, if there is one provided (eg supplied by redux-form or DateTimePicker)
 	*/
-	onTimeInputChange(e) {
-		const { value } = e.target;
-		this.setState(() => ({ value }));
+  onTimeInputChange(e) {
+    const { value } = e.target;
+    // In a time input component, the time is first set as an empty string. We only care about the valid time, so return out.
```

Comment:
What does 'the time is first set as an empty string' mean? In the problem we were seeing, the `value` attribute of the element was already set (e.g. `<input type="time" value="18:00" />`, so it definitely wasn't empty at first. The problem we were seeing was `value` getting set to the empty string _when a user manually entered `0` in the hours field_ - something like `'time inputs are allowed to return an empty string for invalid input - ignore all invalid input'`

sort of nitpick but 'return out' is sort of an ambiguous turn of phrase. 'return immediately' or 'exit early' are a bit more clear.

Reply:
done